### PR TITLE
add option for inserting into header

### PR DESCRIPTION
**Pull request by keertip**

### CODE REVIEW

File: bin/dartdoc.dart [hunk 1]
```diff
@@ -53,7 +53,18 @@ void main(List<String> arguments) {
   List<String> excludeLibraries =
       args['exclude'] == null ? [] : args['exclude'].split(',');
   String url = args['hosted-url'];
-  String footer = _resolveTildePath(args['footer']);
+  String footerFilePath = _resolveTildePath(args['footer']);
```
Comment (sethladd) [inline bin/dartdoc.dart]: Should we write a single function that resolves the Tilde and prints the warning if the file doesn't exist? Looks like two copies of basically the same logic.

File: test/html_generator_test.dart [hunk 2]
```diff
@@ -60,7 +60,7 @@ void main() {
     test('header and footer', () {
       String content = templates.indexTemplate({},
           assumeNullNonExistingProperty: true, errorOnMissingProperty: false);
-      expect(content.contains('<p>User inserted message</p>'), true);
+      expect(content.contains('<p>User inserted message</p>'), false);
```
Comment (sethladd) [inline test/html_generator_test.dart]: for next time: isFalse

or:

expect(content, contains(...))

(the error messages are nicer this way)

File: bin/dartdoc.dart (hunk 1 above)
Comment (keertip) [inline bin/dartdoc.dart]: Yes, we can.

File: test/html_generator_test.dart (hunk 2 above)
Comment (keertip) [inline test/html_generator_test.dart]: Thanks, will do.

**Pull request merged**